contracts/staking/ERC20/StakingERC20.sol

```diff
@@ -1,88 +1,283 @@
  */
 contract StakingERC20 is StakingBase, IStakingERC20 {
+    // TODO when ERC20Voter token is ready add here
+
     using SafeERC20 for IERC20;
 
     constructor(
-        address _stakingToken,
+        IERC20 _stakingToken,
         IERC20 _rewardsToken,
         uint256 _rewardsPerPeriod,
         uint256 _periodLength,
-        uint256 _timeLockPeriod,
         address contractOwner
     )
         StakingBase(
-            _stakingToken,
+            address(_stakingToken),
             _rewardsToken,
             _rewardsPerPeriod,
             _periodLength,
-            _timeLockPeriod,
             contractOwner
         )
     {}
 
     /**
-     * @notice Stake an amount of the ERC20 staking token specified
+     * @notice Stake an amount of ERC20 with a lock period By locking, 
+     * a user cannot access their funds until the lock period is over, but they
+     * receive a higher rewards rate for doing so
+     * @dev A user can call to `unstake` with `exit` as true to access their funds
+     * before the lock period is over, but they will forfeit their rewards
+     * @dev This function and the below `stakeWithoutLock` are intentionally separate for clarity
+     * 
+     * @param amount The amount to stake
+     * @param lockDuration The duration of the lock period
+     */
+    function stakeWithLock(uint256 amount, uint256 lockDuration) external override {
+        _stake(amount, lockDuration);
+    }
+
+    /**
+     * @notice Stake an amount of ERC20 with no lock period. By not locking, a 
+     * user can access their funds any time, but they forfeit a higher rewards rate
+     * @dev This function and the above`stakeWithLock` are intentionally separate for clarity
+     * 
      * @param amount The amount to stake
      */
-    function stake(uint256 amount) external override {
+    function stakeWithoutLock(uint256 amount) external override {
+        _stake(amount, 0);
+    }
+
+    /**
+     * @notice Claim all of the user's rewards that are currently available
+     */
+    function claim() external {
+        // transfer a user their owed rewards + any available pending rewards
+        // if funds are locked, only transfer if they are past lock duration
         Staker storage staker = stakers[msg.sender];
 
-        if (amount == 0) {
-            revert ZeroStake();
+        // Get rewards they've accumulated already as well as any pending rewards from their last timestamp
+        uint256 rewards = _getPendingRewards(staker);
+
+        staker.owedRewards = 0;
+        staker.lastTimestamp = block.timestamp;
+
+        if (staker.unlockedTimestamp != 0 && _getRemainingLockTime(staker) == 0) {
+            // If the above is true, the rewards will have already been accounted for in
+            // the first `_getPendingRewards` call
+            // We only need to update here
+            staker.owedRewardsLocked = 0;
+            staker.lastTimestampLocked = block.timestamp;
```

Comment:
could you explain these 2 lines for me? why do we set owedRewards to 0 and update lastTimestampLocked here during claim?

Reply:
`_getPendingRewards` is `view` so we update state variables here.